ffi/src/prove.rs

```diff
@@ -31,11 +48,51 @@ pub unsafe extern "C" fn create_block_proof(trace_char: *const c_char) -> *const
 pub unsafe extern "C" fn create_chunk_proof(trace_char: *const c_char) -> *const c_char {
     let trace_vec = c_char_to_vec(trace_char);
     let traces = serde_json::from_slice::<Vec<BlockTrace>>(&trace_vec).unwrap();
-    let proof = PROVER
+    let proof = CHUNK_PROVER
         .get_mut()
         .unwrap()
         .gen_chunk_proof(traces.as_slice())
         .unwrap();
     let proof_bytes = serde_json::to_vec(&proof).unwrap();
     vec_to_c_char(proof_bytes)
 }
+
+/// # Safety
+#[no_mangle]
+pub unsafe extern "C" fn add_agg_chunk_trace(trace_char: *const c_char) {
```

Comment:
we should not make a stateful API. You should just create a single API
```
fn create_agg_proof(chunk_trace: *const c_char) -> *const c_char;
```

Reply:
Yes. There is no this API for final FFI (have not finished updated FFI).

Will update as two FFI functions:
- one chunk-trace -> one two-layer-compression proof (as chunk-proof)
- Multiple two-layer-compression proofs -> one aggregation-proof (as batch-proof)

Is it right? Thanks.

Reply:
Updated (in-progress) functions [gen_chunk_proof](https://github.com/scroll-tech/scroll-prover/blob/agg-funs/ffi-refactor/prover/src/aggregator/prover.rs#L136) and [gen_agg_proof](https://github.com/scroll-tech/scroll-prover/blob/agg-funs/ffi-refactor/prover/src/aggregator/prover.rs#L79C12-L79C25) in `agg-funs/ffi-refactor` branch.

Reply:
Update FFI prove to [gen_chunk_proof](https://github.com/scroll-tech/scroll-prover/blob/fc94edf7f55a250b3a51eb298a21715857f8db6d/ffi/src/prove.rs#L58) and [gen_agg_proof](https://github.com/scroll-tech/scroll-prover/blob/fc94edf7f55a250b3a51eb298a21715857f8db6d/ffi/src/prove.rs#L74) in commit [fc94edf](https://github.com/scroll-tech/scroll-prover/pull/160/commits/fc94edf7f55a250b3a51eb298a21715857f8db6d).